src/auth/AuthProvider.tsx

```diff
@@ -0,0 +1,153 @@
+      try {
+        const token = await openidManager.getAccessToken();
+        if (token) {
+          const userInfo = await openidManager.getUserInfo();
+          if (!cancelled) {
+            setState({
+              isAuthenticated: true,
+              user: {
+                id: userInfo.sub || userInfo.id || "unknown",
+                email: userInfo.email || "",
+                name: userInfo.name,
+              },
+              token,
+              isLoading: false,
+              error: null,
+            });
+          }
+        } else {
+          if (!cancelled) setState((prev) => ({ ...prev, isLoading: false }));
+        }
+      } catch (error: any) {
+        if (!cancelled) setState((prev) => ({ ...prev, isLoading: false, error: error.message || String(error) }));
+      }
+    })();
+    return () => { cancelled = true; };
+  }, []);
+
+  const signIn = useCallback(async () => {
+    try {
+      const url = await openidManager.getAuthorizationUrl("login");
+      window.location.href = url.toString();
+    } catch (error: any) {
+      setState((prev) => ({ ...prev, error: error.message || String(error) }));
+    }
+  }, []);
+
+  const register = useCallback(async () => {
+    try {
+      const url = await openidManager.getAuthorizationUrl("registration");
+      window.location.href = url.toString();
+    } catch (error: any) {
+      setState((prev) => ({ ...prev, error: error.message || String(error) }));
+    }
+  }, []);
+
+  const signOut = useCallback(async () => {
+    try {
+      await openidManager.logout();
+      setState({
+        isAuthenticated: false,
+        user: null,
+        token: null,
+        isLoading: false,
+        error: null,
+      });
+    } catch (error: any) {
+      setState((prev) => ({ ...prev, error: error.message || String(error) }));
+    }
+  }, []);
+
+  const refreshToken = useCallback(async () => {
+    try {
+      const token = await openidManager.getAccessToken();
+      setState((prev) => ({ ...prev, token }));
+    } catch (error: any) {
+      setState((prev) => ({ ...prev, error: error.message || String(error) }));
+    }
+  }, []);
+
+  // Optionally: handle OpenID redirect response (should be called after redirect)
+  const handleRedirect = useCallback(async (url: URL) => {
+    setState((prev) => ({ ...prev, isLoading: true }));
+    try {
+      await openidManager.handleRedirectResponse(url);
+      const token = await openidManager.getAccessToken();
+      const userInfo = await openidManager.getUserInfo();
+      setState({
+        isAuthenticated: true,
+        user: {
+          id: userInfo.sub || userInfo.id || "unknown",
+          email: userInfo.email || "",
+          name: userInfo.name,
+        },
+        token,
+        isLoading: false,
+        error: null,
+      });
+    } catch (error: any) {
+      setState((prev) => ({ ...prev, isLoading: false, error: error.message || String(error) }));
```

Comment:
We're directly exposing errors to users this way, which is probably fine for now